src/backend/cdb/cdbquerycontextdispatching.c

```diff
@@ -779,21 +779,24 @@ static void
 RebuildNamespace(QueryContextInfo *cxt)
 {
 
+	MemoryContext oldContext;
 	int len;
 	char buffer[4], *binary;
+	oldContext = MemoryContextSwitchTo(MessageContext);
 	ReadData(cxt, buffer, sizeof(buffer), TRUE);
 
 	len = (int) ntohl(*(uint32 *) buffer);
 	binary = palloc(len);
 	if(ReadData(cxt, binary, len, TRUE))
 	{
-		StringInfoData buffer;
-		initStringInfoOfString(&buffer, binary, len);
-		dfs_address = strdup(buffer.data);
+		dfs_address = pstrdup(binary);
```

Comment:
ok, I thought you used strdup, missed you have changed to pstrdup, this is covered by memorycontext